d8/d8forms/d8forms.module

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/**
+ * @file
+ * Contains d8forms.module..
+ */
+
+use Drupal\Core\Routing\RouteMatchInterface;
+
+define('D8FORMS_MODULE_NAME', 'd8forms');
+
+/**
+ * Implements hook_help().
+ */
+function d8forms_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the d8forms module.
+    case 'help.page.d8forms':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Form examples in D8.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Add ability to node type delete forms to remove existing nodes with node
+ * types.
+ */
+function d8forms_form_node_type_delete_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
+  $build_info = $form_state->getBuildInfo();
+  $form['remove_nodes'] = [
+    '#type' => 'submit',
+    '#access' => (bool) count(d8forms_get_node_type_nodes($build_info['callback_object']->getEntity()
+      ->get('type'))),
+    '#value' => t('Remove all %node_type nodes', [
+      '%node_type' => $build_info['callback_object']->getEntity()->get('name'),
+    ]),
+    '#submit' => ['d8forms_form_node_type_delete_form_removes_nodes_submit'],
+  ];
+}
+
+/**
+ * Form submit handler for "Remove all %node_type nodes" button.
+ *
+ * @see d8forms_form_node_type_delete_form_alter()
+ */
+function d8forms_form_node_type_delete_form_removes_nodes_submit(&$form, \Drupal\Core\Form\FormStateInterface $form_state) {
+  $build_info = $form_state->getBuildInfo();
+  $nids = d8forms_get_node_type_nodes($build_info['callback_object']->getEntity()
+    ->get('type'));
+
+  if (!empty($nids)) {
+    $batch = [
+      'title' => t('Removing all %node_type nodes', [
+        '%node_type' => $build_info['callback_object']->getEntity()->get('name'),
+      ]),
+      'operations' => [
+        [
+          '_d8forms_delete_all_node_type_nodes_batch_operation',
+          [$nids, $build_info['callback_object']->getEntity()->get('name')],
+        ],
+      ],
+      'progress_message' => '',
```

Comment:
Yes, because I would like to hide the original (miss-leading) progress message. My module displays better info than that about the status of the current progress. But this could be simple just NULL, thanks for noticing that.